### PR TITLE
Bug fix for moving average tracking

### PR DESCRIPTION
**Pull request by nathanielpritchard**
1. Added more functionality to the moving average tracking to allow for use of a short width and long width moving average, which I thought worked but upon a closer look did not work.

### CODE REVIEW

File: src/linear_solver_logs/solve_log_ma.jl [hunk 1]
```diff
@@ -62,6 +62,7 @@ mutable struct SEDistInfo
     sigma2::Union{Float64, Nothing}
     omega::Union{Float64, Nothing}
     eta::Float64
+    scaling::Float64
```
Comment (vp314) [inline src/linear_solver_logs/solve_log_ma.jl:65]: This field is not explained in the docstring.

File: src/linear_solver_logs/solve_log_ma.jl [hunk 2]
```diff
@@ -183,17 +184,18 @@ function log_update!(
     if !log.true_res && iter > 0
         # Compute the current residual to second power to align with theory
         # Check if it is one dimensional or block sampling method
-        res::Float64 = eltype(samp[1]) <: Int64 || size(samp[1],2) != 1 ? 
-            log.resid_norm(samp[3])^2 : log.resid_norm(dot(samp[1], x) - samp[2])^2 
+        res::Float64 = log.dist_info.scaling * (eltype(samp[1]) <: Int64 || size(samp[1],2) != 1 ? 
+                                                log.resid_norm(samp[3])^2 : log.resid_norm(dot(samp[1], x) - samp[2])^2)
```
Comment (vp314) [inline src/linear_solver_logs/solve_log_ma.jl:188]: Why is there so much white space added in?

File: src/linear_solver_logs/solve_log_ma.jl [hunk 3]
```diff
@@ -183,17 +184,18 @@ function log_update!(
...
         res = log.resid_norm(A * x - b)^2 
     end
+  
     # Check if MA is in lambda1 or lambda2 regime
     if ma_info.flag
         update_ma!(log, res, ma_info.lambda2, iter)
     else
         #Check if we can switch between lambda1 and lambda2 regime
-        if res < ma_info.res_window[ma_info.idx]
+        if iter == 0 || res <= ma_info.res_window[ma_info.idx]
```
Comment (vp314) [inline src/linear_solver_logs/solve_log_ma.jl:198]: This line needs more explanation

File: src/linear_solver_logs/solve_log_ma.jl [hunk 4]
```diff
@@ -244,6 +246,32 @@ function update_ma!(log::LSLogMA, res::Union{AbstractVector, Real}, lambda_base:
...
+            accum += ma_info.res_window[i]
+            accum2 += ma_info.res_window[i]^2
+        end
+
+        #Update the log variable with the information for this update
+        if mod(iter, log.collection_rate) == 0 || iter == 0
+            push!(log.lambda_hist, ma_info.lambda)
+            push!(log.resid_hist, accum / ma_info.lambda) 
+            push!(log.iota_hist, accum2 / ma_info.lambda) 
+        end
```
Comment (vp314) [inline src/linear_solver_logs/solve_log_ma.jl:274]: This conditional statement needs more commentary. When is this accessed?

**Review by vp314 — CHANGES_REQUESTED** [review]

File: src/linear_solver_logs/solve_log_ma.jl (hunk 1 above)
Comment (nathanielpritchard) [reply src/linear_solver_logs/solve_log_ma.jl:65]: I have added this field. It a parameter to ensure that in expectation the sketched norm-squared of the residual is the same as the norm squared of the residual.

File: src/linear_solver_logs/solve_log_ma.jl (hunk 2 above)
Comment (nathanielpritchard) [reply src/linear_solver_logs/solve_log_ma.jl:188]: At the time I thought it was more clear to have the if else statement within the parentheses aligned. I know see that is not the case and I have removed it.

File: src/linear_solver_logs/solve_log_ma.jl (hunk 3 above)
Comment (nathanielpritchard) [reply src/linear_solver_logs/solve_log_ma.jl:198]: I have clarified that this lean is checking for a monotonic decrease in the sketched residuals

File: src/linear_solver_logs/solve_log_ma.jl (hunk 4 above)
Comment (nathanielpritchard) [reply src/linear_solver_logs/solve_log_ma.jl:274]: I see now that this is really unclear, I have added comments that I think clarify what is going on

File: src/linear_solver_logs/solve_log_ma.jl
Comment (vp314) [inline src/linear_solver_logs/solve_log_ma.jl:1]: 1. Decide on whether or not the `dist_info.sampler` field is necessary given that it is only called twice for initializing the constants.
2. `iterations` in `LSLogMA` should be singular.
3. Need to fast forward on top of master

**Review by vp314 — CHANGES_REQUESTED** [review]

File: src/linear_solver_logs/solve_log_ma.jl
Comment (nathanielpritchard) [reply src/linear_solver_logs/solve_log_ma.jl:1]: I have removed the sampler from the list_info and removed the s

Comment (vp314) [inline src/linear_solver_logs/solve_log_ma.jl:1]: Need default behavior for `sigma2` and `omega` when not in look-up table.

Comment (vp314) [reply src/linear_solver_logs/solve_log_ma.jl:1]: Get rid of `true_res` in `LSLogMA` and create a separate logger for it (create an issue)

**Review by vp314 — CHANGES_REQUESTED** [review]

**Review by vp314 — APPROVED** [review]

**Pull request merged**